docs/kyma-environment-broker/03-04-add-deprovisioning-step.md

```diff
@@ -0,0 +1,158 @@
+# Configure Runtime deprovisioning
+
+The operation of deprovisioning a Runtime consists of several steps. Each step is represented by a file that is responsible for a separate part of cleaning Runtime dependencies. 
+
+All the necessary data required to properly deprovision runtime's dependencies can be fetched from the `DeprovisioningOperation` in the [initialisation](https://github.com/kyma-incubator/compass/blob/master/components/kyma-environment-broker/internal/process/deprovisioning/initialisation.go#L46) step.
+
+In case of a processing failure, every step can be re-launched multiple times. That is why, for each step you should determine a behavior in case of a processing failure. It can either:
+- Return an error, which interrupts the entire deprovisioning process, or 
+- Repeat the entire operation after the specified period. 
+
+## Add deprovisioning step
+
+To add a new deprovisioning step, follow these steps:
+
+1. Create a new file in [this](https://github.com/kyma-incubator/compass/blob/master/components/kyma-environment-broker/internal/process/deprovisioning) directory. 
+
+2. Implement this interface in your deprovisioning step:
+
+    ```go
+    type Step interface {
```

Comment:
I'm wondering, why there's a different step interface? doesn't it make sense if we add another method in current step interface's lifecycle?

Reply:
Maybe that's an overhead right now, but I'd like to have this separated because we don't exactly know how many steps will be implemented in the future and how both structs - `ProvisionOperation` and `DeprovisionOperation` will differ. What's more not every ProvisioningOperation needs a deprovisioning callback.